packages/react-components/src/components/chart/chartOptions/seriesAndYAxis/convertSeriesAndYAxis.ts

```diff
@@ -216,38 +226,92 @@ export const reduceSeriesAndYAxis = (
   {
+    defaultVisualizationType,
+    significantDigits,
     styleSettings,
     axis,
     thresholds,
     performanceMode,
-  }: {
-    styleSettings: StyleSettingsMap;
+  }: Pick<
+    ChartOptions,
+    'defaultVisualizationType' | 'styleSettings' | 'significantDigits'
+  > & {
     thresholds: Threshold[];
     axis?: ChartAxisOptions;
     performanceMode?: boolean;
   }
 ) => {
-  const defaultYAxis: YAXisComponentOption[] = [convertChartYAxis(axis)];
-  const convertedThresholds = convertThresholds(thresholds);
-  const getStyles = getChartStyleSettingsFromMap(styleSettings);
-
-  const { series, yAxis } = datastreams
-    .map((datastream) =>
-      convertSeriesAndYAxis(getStyles(datastream), { performanceMode })(
-        datastream
-      )
-    )
-    .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis });
-
-  if (series.length > 0) {
-    series[0].markArea = convertedThresholds.markArea;
-    series[0].markLine = convertedThresholds.markLine;
-  }
+  const { highlightedDataStreams } = useChartStore(
+    (state) => ({ highlightedDataStreams: state.highlightedDataStreams }),
+    isEqual
+  );
+  const isDataStreamHighlighted = useCallback(
+    (datastream: DataStreamInfo) =>
+      isDataStreamInList(highlightedDataStreams)(datastream),
+    [highlightedDataStreams]
+  );
 
-  return {
-    series,
-    yAxis,
-  };
+  const { hiddenDataStreams } = useChartStore(
+    (state) => ({ hiddenDataStreams: state.hiddenDataStreams }),
+    isEqual
+  );
+  const isDataStreamHidden = useCallback(
+    (datastream: DataStreamInfo) =>
+      isDataStreamInList(hiddenDataStreams)(datastream),
+    [hiddenDataStreams]
+  );
+
+  return useMemo(() => {
+    const defaultYAxis: YAXisComponentOption[] = [convertChartYAxis(axis)];
+    const convertedThresholds = convertThresholds(thresholds);
+
+    const shouldUseEmphasis = highlightedDataStreams.length > 0;
+
+    const { series, yAxis } = datastreams
+      .map((datastream, index) => {
+        const isDatastreamHighlighted = isDataStreamHighlighted(datastream);
+        const emphasis: Emphasis = shouldUseEmphasis
+          ? isDatastreamHighlighted
+            ? 'emphasize'
+            : 'de-emphasize'
+          : 'none';
+        const hidden = isDataStreamHidden(datastream);
+        const chartStylesWithDefaults = convertStyles({
+          styleSettings,
+          significantDigits,
+          defaultVisualizationType,
+          emphasis,
+          hidden,
+        })(datastream);
+
+        return convertSeriesAndYAxis(chartStylesWithDefaults, {
+          performanceMode,
+          index,
+        })(datastream);
+      })
+      .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis });
+
+    if (series.length > 0) {
+      series[0].markArea = convertedThresholds.markArea;
+      series[0].markLine = convertedThresholds.markLine;
+    }
+
+    return {
+      series,
+      yAxis,
+    };
+  }, [
+    datastreams,
```

Comment:
i believe the dependencies change on every render... so not worth adding memo, unless all there are also memorized or use state

Reply:
No, we need this memo to prevent the setOption in useChartConfiguration from being excessively called. I had the same thought, but tested with and without and it is required.